tests/write_and_read_1m_messages_with_multi_partitions_and_compression.rs

```diff
@@ -0,0 +1,91 @@
+use futures::stream::iter;
+use futures::StreamExt;
+use samsa::prelude;
+use samsa::prelude::{
+    BrokerConnection, Compression, ConsumerBuilder, Error, KafkaCode, ProduceMessage,
+    ProducerBuilder, TcpConnection, TopicPartitionsBuilder,
+};
+use std::collections::HashMap;
+
+mod testsupport;
+
+const CLIENT_ID: &str = "all together";
+const CORRELATION_ID: i32 = 1;
+const NUMBER_OF_PARTITIONS: i32 = 10;
+const CHUNK_SIZE: usize = 50_000;
+
+#[tokio::test]
+async fn write_and_read_1m_messages_with_multi_partitions_and_compression() -> Result<(), Box<Error>>
+{
+    let (skip, brokers) = testsupport::get_brokers()?;
+    if skip {
+        return Ok(());
+    }
+    let conn = TcpConnection::new(brokers.clone()).await?;
+    let topic_name = "tester-topic";
+
+    //
+    // Create topic with 1O partitions
+    //
+    let create_res = prelude::create_topics(
+        conn.clone(),
+        CORRELATION_ID,
+        CLIENT_ID,
+        HashMap::from([(topic_name, NUMBER_OF_PARTITIONS)]),
+    )
+    .await?;
+
+    // TopicAlreadyExists is an acceptable error in the instance of this test
+    // aborting prematurely
+    if create_res.topics[0].error_code != KafkaCode::TopicAlreadyExists {
+        assert_eq!(create_res.topics[0].error_code, KafkaCode::None);
+    }
+
+    //
+    // Test producing (writing) for 1 million topics
+    //
+    let stream = iter(0..1_000_000).map(|i| ProduceMessage {
+        topic: topic_name.to_string(),
+        partition_id: i % 4, // should give a number between 0 and 10 each time
```

Comment:
The `%` operator will never yield higher than value on the right, so this one will repeat `0, 1, 2, 3` as `i` increases.

Change 4 to 10 and you will receive the correct values